pkg/validation/validation_test.go

```diff
@@ -398,3 +404,767 @@ func TestGatewayClass(t *testing.T) {
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      "invalid-ip-host-contour",
+				},
+				Spec: operatorv1alpha1.ContourSpec{
+					Namespace: operatorv1alpha1.NamespaceSpec{
+						Name: ns.Name,
+					},
+					GatewayClassRef: pointer.StringPtr("invalid-ip-host-gc"),
+				},
+			},
+			gc: &gatewayv1alpha1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "invalid-ip-host-gc",
+				},
+				Spec: gatewayv1alpha1.GatewayClassSpec{
+					Controller: operatorv1alpha1.GatewayClassControllerRef,
+					ParametersRef: &gatewayv1alpha1.ParametersReference{
+						Group:     operatorv1alpha1.GatewayClassParamsRefGroup,
+						Kind:      "Contour",
+						Name:      "invalid-ip-host-contour",
+						Scope:     pointer.StringPtr("Namespace"),
+						Namespace: pointer.StringPtr(ns.Name),
+					},
+				},
+				Status: newGatewayClassAdmittedStatus(),
+			},
+			gateway: &gatewayv1alpha1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      "invalid-ip-host-gateway",
+				},
+				Spec: gatewayv1alpha1.GatewaySpec{
+					GatewayClassName: "invalid-ip-host-gc",
+					Listeners: []gatewayv1alpha1.Listener{
+						{
+							Hostname: &ip,
+							Port:     gatewayv1alpha1.PortNumber(int32(1)),
+							Protocol: gatewayv1alpha1.HTTPProtocolType,
+						},
+					},
+				},
+			},
+			expected: false,
+		},
+		"invalid wildcard host gateway": {
+			contour: &operatorv1alpha1.Contour{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      "invalid-wildcard-host-contour",
+				},
+				Spec: operatorv1alpha1.ContourSpec{
+					Namespace: operatorv1alpha1.NamespaceSpec{
+						Name: ns.Name,
+					},
+					GatewayClassRef: pointer.StringPtr("invalid-wildcard-host-gc"),
+				},
+			},
+			gc: &gatewayv1alpha1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "invalid-wildcard-host-gc",
+				},
+				Spec: gatewayv1alpha1.GatewayClassSpec{
+					Controller: operatorv1alpha1.GatewayClassControllerRef,
+					ParametersRef: &gatewayv1alpha1.ParametersReference{
+						Group:     operatorv1alpha1.GatewayClassParamsRefGroup,
+						Kind:      "Contour",
+						Name:      "invalid-wildcard-host-contour",
+						Scope:     pointer.StringPtr("Namespace"),
+						Namespace: pointer.StringPtr(ns.Name),
+					},
+				},
+				Status: newGatewayClassAdmittedStatus(),
+			},
+			gateway: &gatewayv1alpha1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      "invalid-wildcard-host-gateway",
+				},
+				Spec: gatewayv1alpha1.GatewaySpec{
+					GatewayClassName: "invalid-wildcard-host-gc",
+					Listeners: []gatewayv1alpha1.Listener{
+						{
+							Hostname: &invalidWildcard,
+							Port:     gatewayv1alpha1.PortNumber(int32(1)),
+							Protocol: gatewayv1alpha1.HTTPProtocolType,
+						},
+					},
+				},
+			},
+			expected: false,
+		},
+		"invalid listener, tls protocol without config": {
```

Comment:
i think this might be legal right now in Contour since we're not terminating TLS with TLSRoute and just doing passthrough for now cc @stevesloka

Reply:
though it is a little counter to the described spec